File: go/hamming/hamming.go
// Package hamming has functionality related to
// computing the Hamming distance for DNA
package hamming

import "errors"

// Distance reports the Hamming distance
// between two strands of DNA
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("Bad input")
	}

	dist := 0

	runesA := []rune(a)
	runesB := []rune(b)

	for pos, runeA := range runesA {
		if runesB[pos] != runeA {
			dist++
		}
	}

	return dist, nil
}

File: csharp/hamming/Hamming.cs
using System;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        if (firstStrand.Length != secondStrand.Length) {
            throw new ArgumentException();
        }

        var score = 0;
        for (var i=0; i < firstStrand.Length; i++) {
            if (firstStrand[i] != secondStrand[i]) {
                score++;
            }
        }

        return score;
    }
}
